app/models/spree/shipment_decorator.rb

```diff
@@ -39,20 +39,29 @@ def rebuild_easypost_shipment
       new_ep_shipment = build_easypost_shipment
       new_ep_rates = new_ep_shipment.rates
 
-      matching_rates = new_ep_rates.select do |sr|
-        ("#{sr.carrier} #{sr.service}" == selected_shipping_rate.name) &&
-          (sr.rate.to_f == selected_shipping_rate.cost) &&
-          (selected_shipping_rate.shipping_method.code.match(/#{sr.carrier_account_id}/))
+      matching_rates = new_ep_rates.select do |ep_shipping_rate|
+        names_match = "#{ep_shipping_rate.carrier} #{ep_shipping_rate.service}" == selected_shipping_rate.name
```

Comment:
rather than assigning local variables, you could write the `select` block as
```
next unless "#{ep_shipping_rate.carrier} #{ep_shipping_rate.service}" == selected_shipping_rate.name
next unless selected_shipping_rate.shipping_method.code.match(/#{ep_shipping_rate.carrier_account_id}/)
if selected_shipping_rate.is_flat_rate
  ep_shipping_rate.rate.to_f == selected_shipping_rate.actual_cost
else
  ep_shipping_rate.rate.to_f == selected_shipping_rate.cost
end